src/main/java/org/elasticsearch/action/admin/indices/seal/SealIndicesResponse.java

```diff
@@ -0,0 +1,124 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.seal;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.indices.SyncedFlushService;
+
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * A response to a seal action on several indices.
+ */
+public class SealIndicesResponse extends ActionResponse implements ToXContent {
+
+    private Set<SyncedFlushService.SyncedFlushResult> results;
+
+    SealIndicesResponse() {
+    }
+
+    SealIndicesResponse(Set<SyncedFlushService.SyncedFlushResult> results) {
+        this.results = results;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        int size = in.readVInt();
+        results = new HashSet<>();
+        for (int i = 0; i < size; i++) {
+            SyncedFlushService.SyncedFlushResult syncedFlushResult = new SyncedFlushService.SyncedFlushResult();
+            syncedFlushResult.readFrom(in);
+            results.add(syncedFlushResult);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeVInt(results.size());
+        for (SyncedFlushService.SyncedFlushResult syncedFlushResult : results) {
+            syncedFlushResult.writeTo(out);
+        }
+    }
+
+    public Set<SyncedFlushService.SyncedFlushResult> results() {
+        return results;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        Map<String, Map<Integer, List<Map.Entry<ShardRouting, SyncedFlushService.SyncedFlushResponse>>>> successfulShards = new HashMap<>();
+        Map<String, Map<Integer, String>> unsuccessfulShards = new HashMap<>();
+        // first, sort everything by index and shard id
+        for (SyncedFlushService.SyncedFlushResult result : results) {
+            if (result.shardResponses().size() > 0) {
+                if (successfulShards.get(result.getShardId().index().name()) == null) {
+                    successfulShards.put(result.getShardId().index().name(), new HashMap<Integer, List<Map.Entry<ShardRouting, SyncedFlushService.SyncedFlushResponse>>>());
```

Comment:
I wonder if it would make sense to use a SortedMap here to get a nice sorted response?